src/NLog/LayoutRenderers/Log4JXmlEventLayoutRenderer.cs

```diff
@@ -340,12 +340,13 @@ private void AppendMdlc(XmlWriter xtw)
 #endif
         }
 
-        private void AppendNdc(XmlWriter xtw)
+        private void AppendNdc(XmlWriter xtw, LogEventInfo logEvent)
         {
             string ndcContent = null;
             if (IncludeNdc)
             {
-                ndcContent = string.Join(NdcItemSeparator, NestedDiagnosticsContext.GetAllMessages());
+                var ndcRenderer = new NdcLayoutRenderer(){ Separator = NdcItemSeparator};
```

Comment:
Instead of allocating a `new NdcLayoutRenderer` for every LogEvent. Could you put the `_ndcRenderer` as member-variable, or extract an internal method for rendering the NDC-string?

Reply:
Ok, I've choosen to use the member variable, because this formating is the main purpose of the NdlcLayoutRenderer and therefore it would be left as an empty wrapper for this method after extracting the logic.

Do you think, the initialization of the fields should be defered until the first use?